gmail/src/services/lead.ts

```diff
@@ -0,0 +1,21 @@
+import { Lead } from "../models/lead";
+import { postJsonRpc } from "../utils/http";
+import { URLS } from "../const";
+import { State } from "../models/state";
+
+/**
+ * Make a RPC call to the Odoo database to create a lead
+ * and return the ID of the newly created record.
+ */
+export function createLead(partnerId: number, emailBody: string, emailSubject: string): number {
```

Comment:
I see you use the "services" way.
The "Odoo" way is to put all those on the "lead" model itself.

So you could have a "create" method on your "lead" model that takes values and returns an instance of the Lead model instead of some id.
Would be cool, I let you see if it's possible / worth it.

Reply:
No, it's a super idea, it will be even easier to add new module :)

Reply:
Good idea :) It will make some model heavy (Partner) but it will be easier to add new model (lead, ticket)